### PR TITLE
Exposed Fail_reload and Skip_reload functionality for conf index load api

### PR DESCRIPTION
**Pull request by suryakumar1024**
Signed-off-by: suryakumar.kumaravelan <suryakumar.kumaravelan@seagate.com>

# Problem Statement
- Conf Fail reload and Skip reload arguments for conf.load() API
- Change "overwrite" to "fail_reload" for consistency. We will have 3 modes	
-fail_reload - Fail on reload
-skip_reload - Dont reload. But Dont generate error. 
-reload - Just reload, i.e.  fail_reload = False, skip_reload = False 
-Default behaviour - fail_reload = True, skip_reload = False

# Design
-  For Bug describe the fix here. 
-  For Feature, Post the link to the solution page on the confluence CORTX Foundation Library 

# Coding 
-  Coding conventions are followed and code is consistent [Y/N]: 
-  Confirm All CODACY errors are resolved [Y/N]: 

# Testing 
- [x] Confirm that Test Cases are added (for both the cases, fix and feature) [Y/N]: 
- [x] Confirm Test Cases cover Happy Path, Non-Happy Path and Scalability [Y/N]: 
- [ ] Confirm Testing was performed with installed RPM [Y/N]:  

# Review Checklist 
  Before posting the PR please ensure
- [x] PR is self reviewed
- [x] Jira is updated
- [x] Check if the description is clear and explained. 
- [x] Check Acceptance Criterion is defined. 
- [x] All the tests performed should be mentioned before Resolving a JIRA. 
- [ ] Verification needs to be done before marked as Closed/Verified 

# Documentation
- [ ] Changes done to WIKI / Confluence page

### CODE REVIEW

File: py-utils/src/utils/conf_store/conf_store.py [hunk 1]
```diff
@@ -193,11 +193,11 @@ def init(**kwargs):
             Conf._machine_id = Conf._conf.machine_id
 
     @staticmethod
-    def load(index: str, url: str):
+    def load(index: str, url: str, **kwargs):
```
Comment (ujjwalpl) [inline py-utils/src/utils/conf_store/conf_store.py:196]: We need to add 3 modes
fail_reload - Fail on reload
	skip_reload - Dont reload. But Dont generate error. 
	reload - Just reload, i.e.  fail_reload = False, skip_reload = False 

Default behaviour - fail_reload = True, skip_reload = False

**Review by ujjwalpl — CHANGES_REQUESTED** [review]

File: py-utils/src/utils/conf_store/conf_store.py (hunk 1 above)
Comment (suryakumar1024) [reply py-utils/src/utils/conf_store/conf_store.py:203]: Updated the code in current patch

File: py-utils/test/conf_store/test_conf_store.py [hunk 2]
```diff
@@ -318,13 +321,31 @@ def test_conf_store_get_machin_id_none(self):
...
+
+    def test_conf_load_skip_reload(self):
+        """ Test conf load fail_reload argument """
+        Conf.load('skip_index', 'json:///tmp/file1.json')
+        expected_lst = Conf.get_keys('skip_index')
+        Conf.load('skip_index', 'toml:///tmp/document.toml', skip_reload=True)
+        out_lst = Conf.get_keys('skip_index')
+        self.assertTrue(True if expected_lst == out_lst else False)
+
+
```
Comment (ujjwalpl) [inline py-utils/test/conf_store/test_conf_store.py:347]: Remove extra lines. Max 2 line space.

**Review by ujjwalpl — CHANGES_REQUESTED** [review]

File: py-utils/src/utils/conf_store/conf_store.py [hunk 3]
```diff
@@ -58,23 +58,30 @@ def load(self, index: str, kvs_url: str, **kwargs):
         Parameters:
         index:     Identifier for the config loaded from the KV Store
         kv_store:  KV Store (Conf Backend)
-        overwrite: When False, it throws exception if index already exists. 
-                   Default: False 
+        fail_reload: When False, it throws exception if index already exists.
+                   Default: False
```
Comment (selvakumaar5496) [inline py-utils/src/utils/conf_store/conf_store.py:62]: a contradiction in comments and code.
fail_reload is set to True but Default value is mentioned as False

**Review by selvakumaar5496 — CHANGES_REQUESTED** [review]

File: py-utils/src/utils/conf_store/conf_store.py (hunk 3 above)
Comment (suryakumar1024) [reply py-utils/src/utils/conf_store/conf_store.py]: Updated current patch

File: py-utils/test/conf_store/test_conf_store.py (hunk 2 above)
Comment (suryakumar1024) [reply py-utils/test/conf_store/test_conf_store.py:362]: Updated in current patch

**Review by ujjwalpl — APPROVED** [review]

**Review by selvakumaar5496 — APPROVED** [review]

File: py-utils/src/utils/conf_store/conf_store.py [hunk 4]
```diff
@@ -58,23 +58,31 @@ def load(self, index: str, kvs_url: str, **kwargs):
         Parameters:
         index:     Identifier for the config loaded from the KV Store
         kv_store:  KV Store (Conf Backend)
-        overwrite: When False, it throws exception if index already exists. 
-                   Default: False 
+        fail_reload: When True, it reloads the conf. into existing index
+                     Else False it throws exception if index already exists.
+                     Default: True
+        skip_reload: When True, it skips reloading a index configuration.
```
Comment (sachinpunadikar) [inline py-utils/src/utils/conf_store/conf_store.py:64]: skip_reload: When True, it skips reloading a index configuration. If True, it overrides fail_reload

File: py-utils/src/utils/conf_store/conf_store.py [hunk 5]
```diff
@@ -58,23 +58,31 @@ def load(self, index: str, kvs_url: str, **kwargs):
         Parameters:
         index:     Identifier for the config loaded from the KV Store
         kv_store:  KV Store (Conf Backend)
-        overwrite: When False, it throws exception if index already exists. 
-                   Default: False 
+        fail_reload: When True, it reloads the conf. into existing index
```
Comment (sachinpunadikar) [inline py-utils/src/utils/conf_store/conf_store.py:61]: fail_reload: When True, and if index already exists, load() throws exception. 
                  When True, and if index do not exists, load() succeeds.
                  When false, irrespective of index status, load() succeeds

**Review by sachinpunadikar — CHANGES_REQUESTED** [review]
Please rectify the comments. Take help of suggestion provided.

**Review by sachinpunadikar — APPROVED** [review]
+1

**Pull request merged**